include/roboteam_utils/containers/static_vector.hpp

```diff
@@ -0,0 +1,248 @@
+     */
+    template<typename T, size_t Cap>
+    class static_vector {
+    public:
+        /**
+         * Contained type
+         */
+        using element_type = T;
+
+        /**
+         * Maximum size of the internal array
+         */
+        constexpr static size_t max_size = Cap;
+
+        /**
+         * Type of container, std::array<T, Cap>
+         */
+        using container_type = std::array<element_type, max_size>;
+
+        /**
+         * Iterator used for begin() and end()
+         */
+        using iterator = element_type *;
+
+        /**
+         * Iterator used for cbegin() and cend()
+         */
+        using const_iterator = const iterator;
+
+        /**
+         * Type of the size (size_t)
+         */
+        using size_type = typename container_type::size_type;
+
+    private:
+        /**
+         * Check whether it's default constructible, std::array requires it.
+         */
+        static_assert(std::is_default_constructible_v<element_type>,
+                      "T has to be default constructible due to the nature of std::array.");
+
+        /**
+         * Check whether it's either copy or move constructible.
+         */
+        static_assert(
+                std::is_nothrow_copy_constructible_v<element_type> || std::is_nothrow_move_assignable_v<element_type>,
+                "T is not noexcept move or copyable.");
+
+        /**
+         * Check whether the contained type is constant, if so we cannot placement-new
+         */
+        static_assert(
+                !std::is_const_v<element_type>, "T cannot be const.");
+
+        /**
+         * Internal array used for storage
+         */
+        std::array<element_type, max_size> _data;
+
+        /**
+         * Current size
+         */
+        size_type _size = 0;
+
+    public:
+        /**
+         * Construction from initializer list.
+         */
+        constexpr explicit static_vector(std::initializer_list<T> data) {
+            for (size_t i = 0; i < data.size(); i++) {
+                if constexpr (std::is_nothrow_move_constructible_v<element_type>) {
+                    this->emplace_back(std::move(data[i]));
+                } else {
+                    this->emplace_back(data[i]);
+                }
+            }
+        }
+
+        static_vector() = default;
+
+        /**
+         * Push back an element (const&)
+         */
+        element_type &push_back(element_type const &data) {
+            return this->emplace_back(data);
+        }
+
+        /**
+         * Push back an element (copy)
+         */
+        element_type &push_back(element_type data) {
+            return this->emplace_back(std::move(data));
+        }
+
+        /**
+         * Emplace back an element
+         * @tparam Parameters passed to ctor
+         */
+        template<typename... Tys>
+        element_type &emplace_back(Tys &&... args) {
+            if (size() >= _data.size()) {
```

Comment:
You have now used the template argument Cap, the static constexpr max_size() and data.size() (where data = array<T,max_size> for the same concept.  I would stick with using max_size probably.